File: app/src/main/java/com/example/inklink/models/ReportedArticles.java
package com.example.inklink.models;

public class ReportedArticles {
    private int id, articleId, userId;
    private String reportType, reportStatus;

    public ReportedArticles() {
        this.reportStatus = "unhandled";
    }

    public int getId() { return id; }

    public void setId(int id) { this.id = id; }

    public int getArticleId() { return articleId; }

    public void setArticleId(int articleId) { this.articleId = articleId; }

    public int getUserId() { return userId; }

    public void setUserId(int userId) { this.userId = userId; }

    public String getReportType() { return reportType; }

    public void setReportType(String reportType) { this.reportType = reportType; }

    public String getReportStatus() { return reportStatus; }

    public void setReportStatus(String reportStatus) { this.reportStatus = reportStatus; }
}

File: app/src/main/java/com/example/inklink/models/ReportedArticles.kt
package com.example.inklink.models

/**
 * This class is a model for 'reported_articles' table and contains all the columns of it.
 */
class ReportedArticles {
    var id: Int? = 0
    var articleId: Int? = 0
    var userId: Int? = 0
    var reportType: String? = null
    var reportStatus: String? = null

    init {
        this.reportStatus = "unhandled"
    }

    override fun toString(): String {
        return """
            ReportedArticle: {
                id: ${this.id},
                articleId: ${this.articleId},
                userId: ${this.userId},
                reportType: ${this.reportType},
                reportStatus: ${this.reportStatus},
            }
        """.trimIndent()
    }
}
